cmd/minikube/cmd/node_add.go

```diff
@@ -109,6 +147,45 @@ func init() {
 	nodeAddCmd.Flags().BoolVar(&cpNode, "control-plane", false, "If set, added node will become a control-plane. Defaults to false. Currently only supported for existing HA (multi-control plane) clusters.")
 	nodeAddCmd.Flags().BoolVar(&workerNode, "worker", true, "If set, added node will be available as worker. Defaults to true.")
 	nodeAddCmd.Flags().BoolVar(&deleteNodeOnFailure, "delete-on-failure", false, "If set, delete the current cluster if start fails and try again. Defaults to false.")
+	// nodeAddCmd.Flags().StringVar(&osType, "os", "linux", fmt.Sprintf("OS of the node to add in the format 'os=OS_TYPE,version=VERSION'. For example, 'os=windows,version=2022'. Valid options: %s (default: linux)", strings.Join(node.ValidOS(), ", ")))
+	nodeAddCmd.Flags().StringVar(&osType, "os", "linux", osTypeLong)
+	// nodeAddCmd.Flags().StringVar(&windowsVersion, "windows-node-version", constants.DefaultWindowsNodeVersion, "The version of Windows to use for the Windows node on a multi-node cluster (e.g., 2019, 2022).")
 
 	nodeCmd.AddCommand(nodeAddCmd)
 }
+
+func validateOS(os string) error {
+	validOptions := node.ValidOS()
+
+	for _, validOS := range validOptions {
+		if os == validOS {
+			return nil
+		}
+	}
+
+	return errors.Errorf("Invalid OS: %s. Valid OS are: %s", os, strings.Join(validOptions, ", "))
+}
+
+// parse the --os flag
```

Comment:
```suggestion
// parseOSFlag parses the --os flag
```
The standard is that Go func comments start with the func name

Reply:
Resolved this!